test/discovery.test.js

```diff
@@ -609,6 +672,27 @@ describe('discoverExportedForeignKeys', function() {
         done(err);
       });
   });
+
+  it('should use custom nameMapper function to `discoverExportedForeignKeys`', function(done) {
+    // Custom version of fake -- need to assert that modelName is being overridden!
+    ds.connector.discoverExportedForeignKeys = function(modelName, options, cb) {
+      modelName.should.equal('INVENTORY');
+      process.nextTick(function() {
+        cb(null, exportedForeignKeys);
+      });
+    };
+    var customMapper = function(type, name) {
+      return name.toUpperCase();
+    };
+    // Table name is intentionally lowercase!
+    ds.discoverExportedForeignKeys('inventory', {
+      nameMapper: customMapper,
+    }, function(err, schemas) {
+      if (err) return done(err);
+      schemas.should.be.eql(exportedForeignKeys);
```

Comment:
nitpick: `schemas` are used in all discover functions, while not all of them return schemas, probably change it to `result`